dm/dm/worker/server.go

```diff
@@ -340,6 +340,10 @@ func (s *Server) observeRelayConfig(ctx context.Context, rev int64) error {
 					}
 					rev = rev1
 					if relaySource == nil {
+						if w := s.getWorker(true); w != nil && w.startedRelayBySourceCfg {
```

Comment:
maybe some retry logic can move in to `w.handleRelayStage`, for retry, jus call handleRelayStage again?

Reply:
don't have to change it in this pr, I'm just making a slight suggestion

Reply:
what's the relation of this function and handleRelayStage?

Reply:
sorry, it's `handleRelayConfig` not `handleRelayStage`
i think from L324 to L377 is retry logic for `handleRelayConfig`, so maybe it's beterr handle retry in `handleRelayConfig`